src/cwe_checker_lib/src/checkers/cwe_78/context/mod.rs

```diff
@@ -27,29 +26,28 @@ pub struct Context<'a> {
     /// A pointer to the representation of the runtime memory image.
     runtime_memory_image: &'a RuntimeMemoryImage,
     /// The reversed control flow graph for the analysis
-    graph: Graph<'a>,
+    graph: Arc<Graph<'a>>,
     /// A pointer to the results of the pointer inference analysis.
     /// They are used to determine the targets of pointers to memory,
     /// which in turn is used to keep track of taint on the stack or on the heap.
     pub pointer_inference_results: &'a PointerInferenceComputation<'a>,
-    /// A map to get the node index of the `BlkStart` node containing a given [`Def`] as the last `Def` of the block.
-    /// The keys are of the form `(Def-TID, Current-Sub-TID)`
-    /// to distinguish the nodes for blocks contained in more than one function.
-    block_start_last_def_map: Arc<HashMap<(Tid, Tid), NodeIndex>>,
-    /// A set containing a given [`Def`] as the first `Def` of the block.
-    /// The keys are of the form `(Def-TID, Current-Sub-TID)`
-    /// to distinguish the nodes for blocks contained in more than one function.
-    block_first_def_set: Arc<HashSet<(Tid, Tid)>>,
+    /// block_first_def_set:
+    ///       A set containing a given [`Def`] as the first `Def` of the block.
+    ///       The keys are of the form `(Def-TID, Current-Sub-TID)`
+    ///       to distinguish the nodes for blocks contained in more than one function.
+    /// block_start_last_def_map:
+    ///       A map to get the node index of the `BlkStart` node containing a given [`Def`] as the last `Def` of the block.
+    ///       The keys are of the form `(Def-TID, Current-Sub-TID)`
+    ///       to distinguish the nodes for blocks contained in more than one function.
+    /// jmp_to_blk_end_node_map:
+    ///       A map to get the node index of the `BlkEnd` node containing a given [`Jmp`].
+    ///       The keys are of the form `(Jmp-TID, Current-Sub-TID)`
+    ///       to distinguish the nodes for blocks contained in more than one function.
+    block_maps: Arc<BlockMaps>,
     /// Maps the TID of an extern symbol to the extern symbol struct.
-    extern_symbol_map: Arc<HashMap<Tid, &'a ExternSymbol>>,
     /// Maps the TID of an extern string related symbol to the corresponding extern symbol struct.
-    string_symbol_map: Arc<HashMap<Tid, &'a ExternSymbol>>,
     /// Maps the TID of an extern symbol that take input from the user to the corresponding extern symbol struct.
-    user_input_symbol_map: Arc<HashMap<Tid, &'a ExternSymbol>>,
-    /// A map to get the node index of the `BlkEnd` node containing a given [`Jmp`].
-    /// The keys are of the form `(Jmp-TID, Current-Sub-TID)`
-    /// to distinguish the nodes for blocks contained in more than one function.
-    jmp_to_blk_end_node_map: Arc<HashMap<(Tid, Tid), NodeIndex>>,
+    symbol_maps: Arc<SymbolMaps<'a>>,
```

Comment:
You forgot to adjust the doc string. The way it is written now is confusing in my opinion.